### ISSUE TITLE
Stubbed results do not maintain consistency between coordinates and lat, long

### ISSUE THREAD

**Issue by derekprior** [alexreisner/geocoder#1258]
The README contains the following code for stubbing Geocoder results:

```ruby
Geocoder.configure(:lookup => :test)

Geocoder::Lookup::Test.add_stub(
  "New York, NY", [
    {
      'coordinates'  => [40.7143528, -74.0059731],
      'address'      => 'New York, NY, USA',
      'state'        => 'New York',
      'state_code'   => 'NY',
      'country'      => 'United States',
      'country_code' => 'US'
    }
  ]
)
```

### Expected behavior

When using the stub above for a test that attempts to Geocode an ActiveRecord model via the `geocode` method should succeed and set `latitude` and `longitude` on the record appropriately.

### Actual behavior

Latitude and longitude remain `nil` despite coordinates being defined in the stub.

The `geocode` method relies on calling `latitude` and `longitude` on the result. `Geocorder::Result::Base` defines `latitude` and `longitude` in terms of `coordinates`, but the test result is essentially a hash with no logic to synchronize lat and long with coordinates.

It seems I have to set `coordinates`, `latitude` and `longitude` in stubs to cover all bases. In my own test helpers for stubbing geocoder responses, I've added the following to compensate for this.

```ruby
  def sync_lat_long_and_coordinates(options)
    if options[:coordinates].present?
      options[:latitude] = options[:coordinates].first
      options[:longitude] = options[:coordinates].last
    elsif options[:latitude].present? && options[:longitude].present?
      options[:coordinates] = [options[:latitude], options[:longitude]]
    end

    options
  end
```

### Environment info

* Geocoder version: 1.4.4
* Rails version:  5.1

**Comment by jheiss** [comment]
Looks like this is related to #1197

**Comment by alexreisner** [comment]
Thanks @derekprior and @jheiss. The intention of the Test lookup/result is to be very generic, and to do very little (since all APIs return different data, it's meant to be very flexible). That being said, ALL results return lat/lon based on the `coordinates` array, so I think there's a good case to be made for implementing `latitude` and `longitude` methods in the Test lookup which will keep things in sync. As I think is clear from my odd response on #1197, I thought it was already like this (oops!).

**Comment by sergiopantoja** [comment]
Another temporary workaround 👍 

```ruby
Geocoder::Lookup::Test.add_stub(
  "New York, NY", [
    {
      'coordinates'  => [40.7143528, -74.0059731],
      'address'      => 'New York, NY, USA',
      'state'        => 'New York',
      'state_code'   => 'NY',
      'country'      => 'United States',
      'country_code' => 'US',
      'latitude'     => 40.7143528,
      'longitude'    => -74.0059731
    }
  ]
)
```

**Comment by CaioBianchi** [comment]
Having the same issue here. Coordinates remain `nil` no matter what I pass in.

**Comment by alexreisner** [comment]
Fixed in `v1.5` branch: b284f71965982ed1c404638012492ca2aee2c379
